apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -308,7 +366,62 @@ defmodule Aecore.Chain.Worker do
     end
   end
 
+
+  ## ============================================================================
+  ##                               Voting
+  ## ============================================================================
+
+
+  ## We are going through each tx in the block and check if it is
+  ## %VotingQuestionTx{} or %VotingAnswerTx{}. If so, we add
+  ## to voting_state new question or update already existing question
+  ## with new answer. After we have updated voting state then we calculate
+  ## the voting result for each questions
+  @spec apply_voting_txs_to_chain(Block.t(), map(), map(), binary()) :: map()
+  defp apply_voting_txs_to_chain(
+    %{txs: txs, header: header} = block, voting_state, chain_states, top_hash) do
+
+    updated_voting_state =
+      Enum.reduce(txs, voting_state,
+        fn(tx, voting_state_acc) ->
+          case tx.data do
+            %VotingTx{data: %VotingQuestionTx{} = question_tx} ->
+              voting_state |>
+                Map.put(SignedTx.hash_tx(tx),
+                  %{data: question_tx, answers: [], result: %{}})
+
+            %VotingTx{data: %VotingAnswerTx{hash_question: hash, answer: answer} = answer_tx} ->
+              Map.put(voting_state_acc, hash,
+                %{voting_state_acc[hash] |
+                  answers: voting_state_acc[hash].answers ++ [answer_tx]})
+
+            _different_tx_data_type_than_voting ->
+              voting_state_acc
+          end
+        end)
+
+    Enum.reduce(updated_voting_state, %{},
+      fn({question_hash, value}, acc) ->
+        result =
+          calculate_vote(question_hash, value.answers, chain_states, top_hash)
+        Map.put(acc, question_hash, %{value | result: result})
+      end)
+  end
+
+  defp calculate_vote(question_hash, answers, chain_states, top_hash) do
```

Comment:
As I understand this calculates the result, maybe rename to `calculate_voting_result`

Reply:
I think this function has a **MAJOR** issue, vote result stakes are still added after the vote is already closed by `close_block_height`, the result should be the stakes snapshot at `close_block_height` if I understand correctly.

Can you add a test-case for this?